middleware/auth.js

```diff
@@ -8,15 +8,14 @@ const { verifyJwt } = require('../infra/jwt');
  */
 const authRequired = async (req, res, next) => {
   const token = req.header('Authorization').replace('Bearer', '').trim();
-  const decoded = verifyJwt(token);
-  const user = await User.findOne({ _id: decoded._id, 'tokens.token': token });
-
-  if (!user) {
+  const decoded = await verifyJwt(token);
+  if(!decoded){
```

Comment:
- Could instead throw and error in the next() middleware here if the token cannot be verified.
- Let the error handler send all the error responses to the client